tests/coverage/extensions.py

```diff
@@ -0,0 +1,284 @@
+# SPDX-License-Identifier: Apache-2.0
+import os
+import yaml
+from ruamel.yaml import YAML
+
+enable_debug = False
+
+
+def error(msg):
+    print(f"ERROR: {msg}")
+
+
+def debug(msg):
+    if enable_debug:
+        print(f"DEBUG: {msg}")
+
+
+type_to_short_type = {
+    "required enumeration": "req",
+    "i8": "i8",
+    "i16": "i16",
+    "i32": "i32",
+    "i64": "i64",
+    "fp32": "fp32",
+    "fp64": "fp64",
+    "string": "str",
+    "binary": "vbin",
+    "boolean": "bool",
+    "timestamp": "ts",
+    "timestamp_tz": "tstz",
+    "date": "date",
+    "time": "time",
+    "interval_year": "iyear",
+    "interval_day": "iday",
+    "uuid": "uuid",
+    "fixedchar<N>": "fchar",
+    "varchar<N>": "vchar",
+    "fixedbinary<N>": "fbin",
+    "decimal<P,S>": "dec",
+    "precision_timestamp<P>": "pts",
+    "precision_timestamp_tz<P>": "ptstz",
+    "struct<T1,T2,...,TN>": "struct",
+    "list<T>": "list",
+    "map<K,V>": "map",
+    "map": "map",
+    "any": "any",
+    "any1": "any1",
+    "any2": "any2",
+    "any3": "any3",
+    "user defined type": "u!name",
+    # added to handle parametrized types
+    "fixedchar": "fchar",
+    "varchar": "vchar",
+    "fixedbinary": "fbin",
+    "decimal": "dec",
+    "precision_timestamp": "pts",
+    "precision_timestamp_tz": "ptstz",
+    "struct": "struct",
+    "list": "list",
+    # added to handle geometry type
+    "geometry": "geometry",
+}
+
+short_type_to_type = {st: lt for lt, st in type_to_short_type.items()}
+
+
+class Extension:
```

Comment:
it isn't clear why we need this stuff. seems like the grammar could handle and move us straight to appropriate objects in many (most?) cases.